clippy_lints/src/methods/mod.rs

```diff
@@ -362,6 +363,29 @@ declare_clippy_lint! {
     "using `ok().expect()`, which gives worse error messages than calling `expect` directly on the Result"
 }
 
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for `.err().expect()` calls on the `Result` type.
+    ///
+    /// ### Why is this bad?
+    /// `.expect_err()` can be called directly to avoid the extra type conversion from `ok()`.
```

Comment:
Shouldn't this say *extra type conversion from `err()`*
```suggestion
    /// `.expect_err()` can be called directly to avoid the extra type conversion
    /// from `err()`.
```